### DISCUSSION TITLE
Add export replay as video

### DISCUSSION THREAD

**Discussion by yesseruser** [ppy/osu#27410] [Ideas] [answered]
This wouldn't be simple to implement, but adding the ability to export replays into videos (e.g. MP4 files) would be nice, especially if it was faster than just opening a screen recording tool and recording the replay yourself.
I know there are replay renderers like [o!rdr](https://ordr.issou.best/) but those don't support Lazer mods and Lazer skins.

**Comment by Gabixel** [comment]
Last time this was discussed in depth, it didn't end very well, as I recall...
See following answers from peppy: [(1)](https://github.com/ppy/osu/discussions/12986#discussioncomment-791141), [(2)](https://github.com/ppy/osu/discussions/12986#discussioncomment-791143) and **[(3)](https://github.com/ppy/osu/discussions/12986#discussioncomment-2626985)**

**Reply by bdach** [reply, marked as answer]
Not happening.

**Reply by yesseruser** [reply]
> Not happening.

How about adding an option to hide replay overlay so I can record without interruptions? Not sure if that's already implemented, though.

**Reply by Joehuu** [reply]
<img width="510" alt="Screenshot 2024-02-28 at 11 08 03 AM" src="https://github.com/ppy/osu/assets/35318437/244ee780-803b-43d6-a14c-2df068f3d3fd">

**Reply by yesseruser** [reply]
@Joehuu Thanks.

**Comment by trustytrojan** [comment]
Hi, I'm not sure if you're still interested in this feature, but I actually implemented it myself in a few days. I just rendered a 1280x720@60fps replay at over 2x speed according to ffmpeg (on my hybrid GPU laptop with integrated Intel Iris Xe doing the video encoding and a dedicated NVIDIA T500 rendering the game). If you'd like to try it, build my [osu](https://github.com/trustytrojan/osu/tree/replay-encoder-attempt) and [osu-framework](https://github.com/trustytrojan/osu-framework/tree/off-screen-capture) branches and let me know how it goes.
